### DISCUSSION TITLE
GitHub Classroom: Peer Review?

### DISCUSSION THREAD

**Discussion by connorferster** [community/community#17533] [GitHub Education]
I just started watching the "Getting Started" videos for GitHub Classroom and I am wondering if there is the ability for the students to do peer review? For example, is there a way to enable each student to have access to the "Feedback Pull Request" form for another (random or assigned) student's work?

I think peer review is a both an efficient and complete part of programming education.

Thank you for any consideration on this feature request!

**Comment by ronanpnc** [comment]
@connorferster 
Following your request, have you found a workaround? It seems that GitHub does not currently offer this feature.

I'm interested to know if you've found any solutions that don't require an expensive license. CodeGrade could work, but it is quite costly
thx !

**Reply by nvaulin** [reply]
As far as I know there are no built-in option for peer-review yet,  but I did it this way:
1) Created public repo for peer review
2) Got the classroom table with students, paths to their private repos and randomly assigned "Nicknames" 
3) Wrote and ran custom script with github CLI which: 
 - Iterates over students
- For each student pulls the code from private repo
- Creates new branch in the peer-review repo and copies the code into that branch (+add +commit). The branch name is the students "Nickname" 
- Finally: push repo with all new branches to github

Then I just send each student:
- His "Nickname"
- Which "Nicknames" he has to review 

Thus each student reviews someones code, can check his own code and comments on it, and also can check out others solutions with reviews. And students don't know who wrote which code (expect for their ones). So the code is anonymous. However, the reviews are not anonymous, but it was ok for me. Non-anonymous reviews = careful reviewers 😄

I can share my custom script if you want. 

Nikita

**Comment by caio-vinicius** [comment]
I'm also searching for this. It would be cool in Classroom to be able to create a peer review between students, specially triggered by students itself.

**Comment by pape98** [comment]
I love Github Classroom. I really do. But my team and I have found over time that it does not seem to be a priority for Github 🥲. So many features we need and no other tools out there that have a GUI. So over time, we developed our own haha (was not the plan). If anyone want to take a look. We've been using it for 3 years now at our school.

[https://classmoji.io/](https://classmoji.io/)

**Reply by pape98** [reply]
@caio-vinicius @connorferster @ronanpnc

**Reply by pape98** [reply]
@nvaulin

**Reply by connorferster** [reply]
Very cool. Thanks for the tag!

**Reply by pape98** [reply]
Of course! Open to ideas or suggestions. Peer review is in our todos.

**Comment by velvethorizon11** [comment]
https://otx.alienvault.com/pulse/6817bbfd22f995c047c7d729
https://otx.alienvault.com/pulse/6817b185294e405c7f93c8f6
https://otx.alienvault.com/pulse/6817af3990dcddd71d4a0e0d
https://otx.alienvault.com/pulse/6817ab44e97fc3badbcd5bbb
